packages/memory/src/index.ts

```diff
@@ -361,27 +362,44 @@ export class Memory extends MastraMemory {
     return result;
   }
 
-  protected mutateMessagesToHideWorkingMemory(messages: MessageType[]) {
+  protected updateMessagesToHideWorkingMemory(messages: MessageType[]): MessageType[] {
     const workingMemoryRegex = /<working_memory>([^]*?)<\/working_memory>/g;
 
-    for (const [index, message] of messages.entries()) {
-      if (typeof message?.content === `string`) {
-        message.content = message.content.replace(workingMemoryRegex, ``).trim();
-      } else if (Array.isArray(message?.content)) {
-        for (const content of message.content) {
-          if (content.type === `text`) {
-            content.text = content.text.replace(workingMemoryRegex, ``).trim();
-          }
+    return messages.reduce<MessageType[]>((acc, message) => {
```

Comment:
Can you simply fhtis a bit by using a for loop or forEach and put some of these desicions in nested if caes and variables,

Reply:
Updated to use for loop instead of reduce